### PR TITLE
Java: update formatting and linting section

### PR DESCRIPTION
**Pull request by CsCherrYY**
This PR extends the `formatting and linting` section and sorts the sections' order as the following: 

1. formatter
2. sonarlint
3. checkstyle 

cc: @nickzhums, @jdneo 

Signed-off-by: Shi Chen chenshi@microsoft.com

### CODE REVIEW

File: docs/java/java-linting.md [hunk 1]
```diff
@@ -13,6 +13,84 @@ MetaDescription: Formatting, linting, and code analysis for Java in Visual Studi
 
 In addition, there are also the [Checkstyle for Java](https://marketplace.visualstudio.com/items?itemName=shengchen.vscode-checkstyle) and [SonarLint](https://marketplace.visualstudio.com/items?itemName=SonarSource.sonarlint-vscode) extensions, which provide features for live linting and code analysis.
 
+## Formatter
+
+Basically, you can use `Format Document` command to format a Java file. If you didn't specify a formatter profile before, the Java file will be formatted with default settings.
```
Comment (nickzhums) [inline docs/java/java-linting.md:18]:
```suggestion
You can use `Format Document` command to format a Java file. If you didn't specify a formatter profile before, the Java file will be formatted using default settings.
```

File: docs/java/java-linting.md [hunk 2]
```diff
@@ -13,6 +13,84 @@ MetaDescription: Formatting, linting, and code analysis for Java in Visual Studi
...
+
+<video autoplay loop muted playsinline controls>
+  <source src="/docs/java/java-linting/formatting.mp4" type="video/mp4">
+</video>
+
+**Applying formatter settings**
+
+You can easily apply formatter settings from an existing formatter profile in Eclipse scheme. For example, you want to apply [Google Style](https://raw.githubusercontent.com/google/styleguide/gh-pages/eclipse-java-google-style.xml) for your Java project, then you can set the following property in `settings.json`:
```
Comment (nickzhums) [inline docs/java/java-linting.md:26]:
```suggestion
You can easily apply formatter settings from an existing formatter profile in Eclipse scheme. For example, if you want to apply [Google Style](https://raw.githubusercontent.com/google/styleguide/gh-pages/eclipse-java-google-style.xml) for your Java project, then you can set the following property in `settings.json`:
```

File: docs/java/java-linting.md [hunk 3]
```diff
@@ -13,6 +13,84 @@ MetaDescription: Formatting, linting, and code analysis for Java in Visual Studi
...
+
+```json
+"java.format.settings.profile": "GoogleStyle",
+```
+
+After setting the formatter profile, `Format Document` command will use the specific profile to format your Java files.
+
+**Editing formatter settings**
+
+[Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack) provides a friendly editor to help the users edit an existing formatter profile. You can open it in command `Java: Open Java Formatter Settings with Preview`. In the editor, you can change the formatter settings and preview the effects. After saving the current editor, the changes will be saved to the formatter profile.
```
Comment (nickzhums) [inline docs/java/java-linting.md:42]:
```suggestion
[Extension Pack for Java](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack) provides a convenient way to edit and preview an existing formatter profile. You can use this feature using command `Java: Open Java Formatter Settings with Preview`. In the editor, you can change the formatter settings and preview the effects. After saving the current editor, the changes will be saved to the formatter profile.
```

File: docs/java/java-linting.md [hunk 4]
```diff
@@ -13,6 +13,84 @@ MetaDescription: Formatting, linting, and code analysis for Java in Visual Studi
...
+
+You can also undo and redo changes.
+
+<video autoplay loop muted playsinline controls>
+  <source src="/docs/java/java-linting/formatting-undoredo.mp4" type="video/mp4">
+</video>
+
+## SonarLint
+
+The [SonarLint](https://marketplace.visualstudio.com/items?itemName=SonarSource.sonarlint-vscode) extension lets you detect bugs and vulnerabilities as you write code in VS Code. Java is one of the languages supported, and the extension will run in the background and highlight source code that poses a quality or security concern.
```
Comment (nickzhums) [inline docs/java/java-linting.md:64]:
```suggestion
The [SonarLint](https://marketplace.visualstudio.com/items?itemName=SonarSource.sonarlint-vscode) extension lets you detect bugs and vulnerabilities as you write code in VS Code. Java is one of the languages supported. The extension runs in the background and highlights source code that poses a quality or security concern.
```

File: docs/java/java-linting.md [hunk 5]
```diff
@@ -13,6 +13,84 @@ MetaDescription: Formatting, linting, and code analysis for Java in Visual Studi
...
+
+<video autoplay loop muted playsinline controls>
+  <source src="/docs/java/java-linting/sonarlint.mp4" type="video/mp4">
+</video>
+
+Issues found in the opened file can also be reviewed through the Problems panel of VS Code. When applicable, secondary code locations are mentioned so you can understand where the issue originates from (for example, the code path that led to a bug).
+
+**Rule documentation and remediation guidance**
+
+For any issue detected, SonarLint provides full documentation about the rule that was violated, and the coding best practice it relates to. This lets you understand why an issue is raised, and most importantly how to best fix it.
```
Comment (nickzhums) [inline docs/java/java-linting.md:78]:
```suggestion
For any issues detected, SonarLint provides full documentation about the rule that was violated, and the best coding practice it relates to. This allows you to understand why an issue is raised, and how to fix it
```

**Comment by nickzhums** [comment]
please fix the comments and rest of the change looks good, thanks!

File: docs/java/java-linting.md (hunk 3 above)
Comment (CsCherrYY) [reply docs/java/java-linting.md:42]: Would `use this feature via` be better?

**Pull request merged**